distributed/protocol/serialize.py

```diff
@@ -596,6 +597,25 @@ def _deserialize_bytearray(header, frames):
         return bytearray().join(frames)
 
 
+@dask_serialize.register(array)
+def _serialize_array(obj):
+    header = {"typecode": obj.typecode}
+    frames = [memoryview(obj)]
+    return header, frames
+
+
+@dask_deserialize.register(array)
+def _deserialize_array(header, frames):
+    a = array(header["typecode"])
+    for f in map(memoryview, frames):
+        try:
+            f = f.cast("B")
+        except TypeError:
+            f = f.tobytes()
```

Comment:
This is needed because `array.frombytes(...)` expects that [`itemsize == 1`]( https://github.com/python/cpython/blob/580fbb018fd0844806119614d752b41fc69660f9/Modules/arraymodule.c#L1597 ). So we perform a cast to `uint8`, which meets this requirement. If that fails, we copy to `bytes`, which also meets this requirement.